src/main/java/salesforce/ui/pages/tables/Table.java

```diff
@@ -0,0 +1,218 @@
+    private Map<String, Integer> columnTitlesIndex;
+    private Map<Integer, String> columnTitles;
+    private Record record;
+    private List<Record> records;
+
+    /**
+     * BaseElement constructor requires a base CSS locator for the table element and the name of the table.
+     *
+     * @param webDriverManager to be managed for the webElementActions
+     */
+    public Table(final WebDriverManager webDriverManager, final String baseLocator, final String featureName) {
+        super(webDriverManager);
+        this.baseLocator = baseLocator;
+        this.featureName = featureName;
+        columnTitlesIndex = new HashMap<String, Integer>();
+        columnTitles = new HashMap<Integer, String>();
+        records = new ArrayList<>();
+        rowsSelector = baseLocator.concat(" tbody tr");
+        rowSelector = baseLocator.concat(" tbody tr:nth-child(%d)");
+        columnSelector = rowSelector.concat(">*:nth-child(%d)");
+        refreshTable();
+    }
+
+    /**
+     * Loads column titles and records count.
+     */
+    public void refreshTable() {
+        fillColumnTitles();
+        getRecordsNumber();
+        records = new ArrayList<>();
+    }
+
+    /**
+     * Gets the titles from the table and maps them with their index for the XML document.
+     */
+    private void fillColumnTitles() {
+        List<WebElement> titleElements = webElementAction.getElements(baseLocator.concat(columnTitleLocator));
+        for (int i = 0; i < titleElements.size(); i++) {
+            columnTitlesIndex.put(titleElements.get(i).getAttribute(columnNameAttributePartialSelector), i + 1);
+            columnTitles.put(i + 1, titleElements.get(i).getAttribute(columnNameAttributePartialSelector));
+        }
+        System.out.print("\n");
+    }
+
+    /**
+     * Sets the number of record results.
+     */
+    private void getRecordsNumber() {
+        this.recordNumber = webElementAction.getElements(rowsSelector).size();
+    }
+
+    /**
+     * Gets a record containing the given value.
+     *
+     * @param fieldValue represents the column value
+     * @return a Record
+     */
+    public Record getRecordByValueForColumn(final String fieldValue, final String columnName) {
+        return getRecordByValue(fieldValue, columnTitlesIndex.get(columnName));
+    }
+
+    /**
+     * Gets a record containing the given value.
+     *
+     * @return a Record
+     */
+    public Record getRecordByName(final String nameValue) {
+        return getRecordByValue(nameValue, nameColumnIndex);
+    }
+
+    /**
+     * Gets a record containing the given value.
+     *
+     * @return a Record
+     */
+    private Record getRecordByValue(final String fieldValue, final int columnIndex) {
+        for (int row = 1; row <= recordNumber; row++) {
+            String fieldValueFound = "";
+            if (getFieldValue(String.format(columnSelector, row, columnIndex)) != null) {
+                if (fieldValueFound.equals(fieldValue)) {

```

Comment:
It would be better only one if
if ((getFieldValue(String.format(columnSelector, row, columnIndex)) != null) & fieldValueFound.equals(fieldValue) ) 
or something like that

Reply:
Done, good observation.